nimbus/src/persistence.rs

```diff
@@ -212,7 +212,29 @@ impl Database {
         let path = std::path::Path::new(path.as_ref()).join("db");
         log::debug!("Database path: {:?}", path.display());
         fs::create_dir_all(&path)?;
-        let rkv = rkv_new(&path)?;
+        let rkv = match rkv_new(&path) {
+            Ok(rkv) => Ok(rkv),
+            Err(rkv_error) => {
+                match rkv_error {
+                    // For some errors we just delete the DB and start again.
+                    StoreError::DatabaseCorrupted | StoreError::FileInvalid => {
+                        // On one had this seems a little dangerous, but on
+                        // the other hand avoids us knowing about the
+                        // underlying implementation (ie, how do we know what
+                        // files might exist in all cases?)
+                        log::warn!(
+                            "Database at '{}' appears corrupt - removing and recreating",
+                            path.display()
+                        );
+                        fs::remove_dir_all(&path)?;
+                        fs::create_dir_all(&path)?;
+                        rkv_new(&path)
```

Comment:
What happens if this also fails?

Reply:
I suppose it just returns the result then...

Reply:
Yeah, the caller will see that failure.